vault/migration-pass.sh

```diff
@@ -1,69 +1,61 @@
 fi
 if [[ -z $VAULT_ADDR ]]; then
-  echo "Please set the varaible $VAULT_ADDR"
+  echo "Please set the variable $VAULT_ADDR"
   exit 1
 fi
 
 if [[ -z "${*}" ]]; then
-    echo "Usage: migration-pass.sh <pass_path> <vault_path>"
+    echo "Usage: migration-pass.sh <vault_path> <pass_path> <other_pass_path>"
     echo
-    echo " - pass_path: path of the secret in the password-store"
     echo " - vault_path: path of the secret in vault"
-    echo 
-    echo "Ex: migration-pass.sh fleet/waku/test/nodekeys waku/test/nodekeys"
-    echo "Or"
-    echo "migration-pass.sh fleet/waku/test/db/nim-waku waku/test/db/nim-waku"
-    echo 
+    echo " - pass_paths: list of path to secret in the password-store, can be a directory to copy all file under it"
+    echo
+    echo "Ex: migration-pass.sh waku/test/nodekeys services/airflow/admin/ services/airflow/fernet-key"
     exit 1
 fi
 
-PASS_PATH="$1"
-VAULT_PATH="$2"
+VAULT_PATH="${1}"
+shift
+
 VAULT="secret"
 
 if [[ -z $PASSWORD_STORE_DIR ]]; then
   echo "env var PASSWORD_STORE_DIR not set, using default location ~/.password-store"
   PASSWORD_STORE_DIR=$HOME/.password-store
 fi
 
-declare -a params
-
-
-if [[ -f "$PASSWORD_STORE_DIR/$PASS_PATH.gpg" ]]; then
-  echo "Reading file $PASS_PATH"
-  field_name=$(basename $PASS_PATH)
-  echo $field_name
-  jq_param=".\"${field_name/./_}\"=\"$(pass $PASS_PATH)\""
-  echo $jq_param
-  params=$(echo "{}" | jq "${jq_param}")
-fi
-
-
-if [[ -d "$PASSWORD_STORE_DIR/$PASS_PATH" ]]; then
-  echo "Reading all file in directory $PASS_PATH"
-  params="{}"
-  files=$(ls "$PASSWORD_STORE_DIR/$PASS_PATH/")
-  while IFS= read -r file; do
-    echo "File name ${file}"
-    if [[ -f "$PASSWORD_STORE_DIR/$PASS_PATH/$file" ]]; then
-      field_name=${file/.gpg/}
-      jq_query=".\"${field_name/./_}\"=\"$(pass $PASS_PATH/${field_name})\""
-      echo 
-      #echo ${jq_query/\n/\\n}
-      echo 
-      params=$(echo $params | jq "${jq_query}")
-    fi
-    echo
-    echo "params: $params"
-  done <<< "$files"
-fi
+params="{}"
+for PASS_PATH in $@; do
+  if [[ -f "$PASSWORD_STORE_DIR/$PASS_PATH.gpg" ]]; then
+    echo "Reading file $PASS_PATH"
+    field_name=$(basename $PASS_PATH)
+    echo "Copying secret from $field_name"
+    jq_param=".\"${field_name/./_}\"=\"$(pass $PASS_PATH)\""
+    params=$(echo "${params}" | jq "${jq_param}")
+  fi
+
+  if [[ -d "$PASSWORD_STORE_DIR/$PASS_PATH" ]]; then
+    echo "Reading all file in directory $PASS_PATH"
+    files=$(ls "$PASSWORD_STORE_DIR/$PASS_PATH/")
+    while IFS= read -r file; do
+      if [[ -f "$PASSWORD_STORE_DIR/$PASS_PATH/$file" ]]; then
+        echo "Copying secret from ${file}"
+        field_name=${file/.gpg/}
+        jq_query=".\"${field_name/./_}\"=\"$(pass $PASS_PATH/${field_name})\""
+        params=$(echo $params | jq "${jq_query}")
+      else
+        echo "Ignoring the sub directory ${file}"
+      fi
+    done <<< "$files"
+  fi
+done
 
 echo "$params" | vault kv put -mount=${VAULT} ${VAULT_PATH} -
```

Comment:
I wonder if we should maybe remove secrets from `pass` too. Or maybe just move them to an `archived` folder.

Reply:
i can add a parameter to indicate to remove the secrets, the git history provide the archive mecanism.

Reply:
It's less visible, but true.

Reply:
Pro  of moving to an archive directory: the old secret will still be accessible if someone after changing it gpg key, if the repo is rencrypted).
Pro of removing the secret and using git as archive: it will reduce the number of secret